scripts/deterministic/DeployScroll.s.sol

```diff
@@ -1011,262 +1011,381 @@ contract DeployScroll is DeterminsticDeployment {
-            notnull(L1_MULTIPLE_VERSION_ROLLUP_VERIFIER_ADDR),
-            MAX_TX_IN_CHUNK
-        );
+        if (getInitializeCount(L1_SCROLL_CHAIN_PROXY_ADDR) == 0) {
+            ScrollChain(L1_SCROLL_CHAIN_PROXY_ADDR).initialize(
+                notnull(L1_MESSAGE_QUEUE_PROXY_ADDR),
+                notnull(L1_MULTIPLE_VERSION_ROLLUP_VERIFIER_ADDR),
+                MAX_TX_IN_CHUNK
+            );
+        }
 
-        ScrollChain(L1_SCROLL_CHAIN_PROXY_ADDR).addSequencer(L1_COMMIT_SENDER_ADDR);
-        ScrollChain(L1_SCROLL_CHAIN_PROXY_ADDR).addProver(L1_FINALIZE_SENDER_ADDR);
+        if (!ScrollChain(L1_SCROLL_CHAIN_PROXY_ADDR).isSequencer(L1_COMMIT_SENDER_ADDR)) {
+            ScrollChain(L1_SCROLL_CHAIN_PROXY_ADDR).addSequencer(L1_COMMIT_SENDER_ADDR);
+        }
+        if (!ScrollChain(L1_SCROLL_CHAIN_PROXY_ADDR).isProver(L1_FINALIZE_SENDER_ADDR)) {
+            ScrollChain(L1_SCROLL_CHAIN_PROXY_ADDR).addProver(L1_FINALIZE_SENDER_ADDR);
+        }
     }
 
     function initializeL2GasPriceOracle() private {
-        L2GasPriceOracle(L2_GAS_PRICE_ORACLE_PROXY_ADDR).initialize(
-            21000, // _txGas
-            53000, // _txGasContractCreation
-            4, // _zeroGas
-            16 // _nonZeroGas
-        );
-
-        L2GasPriceOracle(L2_GAS_PRICE_ORACLE_PROXY_ADDR).updateWhitelist(L1_WHITELIST_ADDR);
+        if (getInitializeCount(L2_GAS_PRICE_ORACLE_PROXY_ADDR) == 0) {
+            L2GasPriceOracle(L2_GAS_PRICE_ORACLE_PROXY_ADDR).initialize(
+                21000, // _txGas
+                53000, // _txGasContractCreation
+                4, // _zeroGas
+                16 // _nonZeroGas
+            );
+        }
+        if (L2GasPriceOracle(L2_GAS_PRICE_ORACLE_PROXY_ADDR).whitelist() != L1_WHITELIST_ADDR) {
+            L2GasPriceOracle(L2_GAS_PRICE_ORACLE_PROXY_ADDR).updateWhitelist(L1_WHITELIST_ADDR);
+        }
     }
 
     function initializeL1MessageQueue() private {
-        L1MessageQueueWithGasPriceOracle(L1_MESSAGE_QUEUE_PROXY_ADDR).initialize(
-            notnull(L1_SCROLL_MESSENGER_PROXY_ADDR),
-            notnull(L1_SCROLL_CHAIN_PROXY_ADDR),
-            notnull(L1_ENFORCED_TX_GATEWAY_PROXY_ADDR),
-            notnull(L2_GAS_PRICE_ORACLE_PROXY_ADDR),
-            MAX_L1_MESSAGE_GAS_LIMIT
-        );
-
-        L1MessageQueueWithGasPriceOracle(L1_MESSAGE_QUEUE_PROXY_ADDR).initializeV2();
+        if (getInitializeCount(L1_MESSAGE_QUEUE_PROXY_ADDR) == 0) {
+            L1MessageQueueWithGasPriceOracle(L1_MESSAGE_QUEUE_PROXY_ADDR).initialize(
+                notnull(L1_SCROLL_MESSENGER_PROXY_ADDR),
+                notnull(L1_SCROLL_CHAIN_PROXY_ADDR),
+                notnull(L1_ENFORCED_TX_GATEWAY_PROXY_ADDR),
+                notnull(L2_GAS_PRICE_ORACLE_PROXY_ADDR),
+                MAX_L1_MESSAGE_GAS_LIMIT
+            );
+        }
+        if (
+            getInitializeCount(L1_MESSAGE_QUEUE_PROXY_ADDR) == 0 || getInitializeCount(L1_MESSAGE_QUEUE_PROXY_ADDR) == 1
```

Comment:
Yes, this is a trick one because of the workflow of forge. 
When forge process the deployment script, it will save the transaction to broadcast file first, and after finished processing the whole deployment script, then it will send transactions in broadcast file to RPC in a row.
Specifically, in this scenario, when we run `getInitializeCount`, the  `initializev1` transaction hasn't really been send to RPC, so it should be 0. And for case `getInitializeCount=1`, this is to enable resume if deployment interrupts after transaction `initializev1`.